en/basic/state.md

```diff
@@ -0,0 +1,103 @@
+# State
+
+"State" should be stored whithin a JavaScript `Object Tree` in the Noflux application. The state should be:
+
+* Immutable
+* Globally unique（single source of truth）
+* Serializable（not required）
+
+> Noflux and Redux have the same concept of state management.
+
+## Immutable
+
+The immutable state means that you shouldn't modify but create a copy of the state. So it could be convenient to make hot reloading, snapshot and so on.
+
+There are many ways to implement an "immutable object" such as using [Immutable.js](https://facebook.github.io/immutable-js/), `Object.assign` in ES2015 and spread operator(`...`) in ES2016. Here is an example:
+
+```js
+const originObj = {
+  a: 1,
+  b: '2',
+  c: { v: 3},
+};
+const newObj = { ...originObj, d: 4};
+```
+
+That object clone implementation is widely used in `Redux` but would be complicated when modifying deeply nested state, for example [here](http://redux.js.org/docs/recipes/reducers/ImmutableUpdatePatterns.html#correct-approach-copying-all-levels-of-nested-data) or this sample:
+
+```js
+const originObj = {
+  a: {
+    b: {
+      c: [1, 2, 3],
+      d: 4,
+    },
+  },
+};
+const newObj = {
+  ...originObj,
+  a: {
+    ...originObj.a,
+    b: {
+      ...originObj.b,
+      d: 3,
+    },
+  },
+};
+```
+
+Changing deeply nested state is simple in the Noflux application by using `state.get` and `state.set`:
+
+```js
+import { state } from '@noflux/react';
+state.set({
+  a: {
+    b: {
+      c: [1, 2, 3],
+      d: 4,
+    },
+  },
+});
+state.set('a.b.d', 5);  // same as clone(originObj).a.b.d = 3
+state.get();            // return the modified object
+```
+
+Noflux provices immutable operators for not only `Object` but also `Array`. For example, `state.get('a.b.c.1')` will return `2`. The string describing the operation path in arguemnts is called "path description string" (more usage [here](../advanced/state.md#path)).
+
+> The implementation of `state` is in `@noflux/state` package ([NPM](https://www.npmjs.com/package/@noflux/state), [Github](https://github.com/nofluxjs/noflux-state)) which provides state management and event listening for Noflux.
+
+## single source of truth
+
+Noflux the global unique state, the **single source of truth** one, which is why you don't need to use `new` to create `state` but directly import it.
```

Comment:
Noflux global unique state, the **single source of truth** one, make it no need to use `new` to create `state` but directly import it.